Dockerfile

```diff
@@ -0,0 +1,56 @@
+# 1: Build the exe
+FROM rust:1.56 as builder
+
+# WORKDIR /usr/src
+
+# RUN USER=root cargo new agora
+
+# COPY Cargo.toml Cargo.lock /usr/src/agora/
+
+WORKDIR /usr/src/agora
+
+# Prepare for static linking
+# RUN apt-get update && \
+#     apt-get dist-upgrade -y && \
+#     apt-get install -y musl-tools && \
+#     rustup target add x86_64-unknown-linux-musl
+
+# Download and compile Rust dependencies (and store as a separate Docker layer)
+
+# copy local dependencies
+# COPY ./agora-lnd-client ./agora-lnd-client
+# COPY ./lnd-test-context ./lnd-test-context
+# COPY ./bin ./bin
+# build dependencies
+# RUN cargo build --release
+
+# Build the exe using the actual source code
+COPY . ./
+RUN cargo build --release
```

Comment:
Can we do `cargo install --path .`? If we do that, it will be installed to some system directory, and we then don't need to copy the binary to run it.

Reply:
we do need to copy it, because `builder` and `final` are two different containers/systems. they wont share any sys directories.

Reply:
Ahhh, gotcha. My lack of docker knowledge is showing.